src/jit/nodeinfo.h

```diff
@@ -32,6 +32,7 @@ class TreeNodeInfo
         isTgtPref             = false;
         regOptional           = false;
         definesAnyRegisters   = false;
+        internalRegDifferentFromTarget = false;
```

Comment:
Suggestion: it might be good to name this `isInternalRegDelayFree` to highlight its similarity (in both requirement, i.e. that it differ from the target, an in implementation.